.github/workflows/ci.yml

```
@@ -0,0 +1,100 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - 'amp/main'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-natives:
+    name: Build Native - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - target: linux_x86_64
+            os: ubuntu-latest
+          - target: mac_x86_64
+            os: macos-latest
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 17
+      - name: Fetch deps
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 8.1
+          cache-read-only: ${{ !startsWith(github.ref, 'refs/heads/amp/main') }}
+          arguments: -I gradle/support/fetchDependencies.gradle init
```

I wonder if this could be sped up with parallelization.
```suggestion
          arguments: --parallel -I gradle/support/fetchDependencies.gradle init
```

---

Also maybe parallelize the other gradle commands?